Classes/Issues/Merge/IssueMergeSectionController.swift

```diff
@@ -65,13 +65,25 @@ MergeButtonDelegate {
         var viewModels = [ListDiffable]()
 
         if object.contexts.count > 0 {
-            let success = object.contexts.reduce(true, { $0 && $1.state == .success })
-            viewModels.append(IssueMergeSummaryModel(
-                title: success ?
-                    NSLocalizedString("All checks passed", comment: "")
-                    : NSLocalizedString("Some checks failed", comment: ""),
-                state: success ? .success : .failure
-            ))
+            let states = object.contexts.map { $0.state }
```

Comment:
As this function is a bit bulky at the moment, I wasn't sure if we would want to pull this out. Thoughts?

Reply:
It's only used here, so not the end of the world. But maybe at the very least we pull it all out into a function? Bonus points if we can find a way to make this stateless inputs/outputs and unit test it!

Reply:
b1cb3c2